src/app/playground/actions/addDevUser.ts

```diff
@@ -0,0 +1,51 @@
+"use server";
+import { Tables } from "@/lib/db";
+import { Shift, UserRole, UserStatus } from "@/constant/constant";
+import { getSupabaseClient } from "@/utils/supabase/supabaseClient";
+import { formatPhoneNumber } from "@/utils/formatPhoneNumber";
+import { DevUserType } from "../components/AddDevUserForm";
+
+export const addDevUser = async (values: DevUserType) => {
```

Comment:
Just a question. Why can't we use `createUser` function here? This would become a test as well to check if we're using that correctly.

Reply:
Sir, there are many reasons:
we are generating a password in `createUser`, but in the case of a `devUser`, we type the password manually.We are using the infer type that depends on addUserSchema that will make issue in type
If we use `createUser`, we will need to handle many conditions, which will make the code complex and messive